src/ethereum_watcher.rs

```diff
@@ -1,43 +1,528 @@
+                format!("Failed to check state contract commits: {e}"),
+                watch_config.invalid_state_commit_alert.alert_level.clone(),
+            );
+            send_action(
+                &action_sender,
+                watch_config.invalid_state_commit_alert.alert_action.clone(),
+                Some(watch_config.invalid_state_commit_alert.alert_level.clone()),
+            );
+            return;
+        },
+    };
+
+    for hash in hashes {
+        match fuel_chain.verify_block_commit(&hash).await {
+            Ok(valid) => {
+                if !valid {
+                    send_alert(
+                        &alert_sender,
+                        String::from("Invalid commit was made on the state contract"),
+                        format!(
+                            "An invalid commit was made on the state contract. Hash: {}", hash,
+                        ),
+                        watch_config.invalid_state_commit_alert.alert_level.clone(),
+                    );
+                    send_action(
+                        &action_sender,
+                        watch_config.invalid_state_commit_alert.alert_action.clone(),
+                        Some(watch_config.invalid_state_commit_alert.alert_level.clone()),
+                    );
+                }
+            }
+            Err(e) => {
+                send_alert(
+                    &alert_sender,
+                    String::from("Failed to check fuel chain state commit contract"),
+                    format!("Failed to check fuel chain state commit: {}", e),
+                    watch_config.invalid_state_commit_alert.alert_level.clone(),
+                );
+                send_action(
+                    &action_sender,
+                    watch_config.invalid_state_commit_alert.alert_action.clone(),
+                    Some(watch_config.invalid_state_commit_alert.alert_level.clone()),
+                );
+            }
+        }
+    }
+    *last_commit_check_block = match ethereum_chain.get_latest_block_number().await {
+        Ok(block_num) => block_num,
+        Err(_) => *last_commit_check_block,
+    };
+}
+
+async fn check_base_asset_deposits(
+    portal_contract: &Arc<dyn PortalContractTrait>,
+    action_sender: UnboundedSender<ActionParams>,
+    alert_sender: UnboundedSender<AlertParams>,
+    watch_config: &EthereumClientWatcher,
+    last_commit_check_block: &u64,
+) {
+    for portal_deposit_alert in &watch_config.portal_deposit_alerts {
+        if portal_deposit_alert.alert_level == AlertLevel::None {
+            continue;
+        }
+
+        let time_frame = portal_deposit_alert.time_frame;
+        let amount = match portal_contract.get_base_amount_deposited(
+            time_frame,
+            *last_commit_check_block,
+        ).await {
+            Ok(amt) => {
+                println!("Ethereum Chain: Total Base Asset Deposited {} for time frame {}",
+                            amt, time_frame);
```

Comment:
Just noticing a lot of `println`s. I imagine these are for debugging purposes. Perhaps they can be changed to DEBUG logs/alerts?

Reply:
They initially had them as such to show that its running, we can ask during the meeting takes 2 seconds to remove